tools/eslint-rules/snake-case-filenames.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * @fileoverview enforces snake_case for filenames.
+ * @author Shobhit Chittora <chittorashobhit@gmail.com>
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  create: function(context) {
+    return {
+      'Program:exit': function(node) {
+        var filename = context.getFilename();
+        context.report(node, `Filename ${filename} does 
+                              not match the naming convention.`);
```

Comment:
And probably the error message can be improved here. More like `Filename ${filename} should be in snake case.` What do you say? :)